README.md
# Moderation

**INACTIVE NOTICE: This plugin is unsupported by WPMUDEV, we've published it here for those technical types who might want to fork and maintain it for their needs.**

## Moderation lets you manage comments and user generated content from across your entire network from one place–just like the big online news sites.

### Classy Content

This powerful plugin allows users to report inappropriate posts and comments, and sends flagged content to one convenient location where designated moderators can send a warning or even completely remove a blog or user.


### A Complete Solution

Configure a massive range of messages and options including which users you want to be moderators, whether they can remove users or blogs, different reasons for reporting different styles of content and notes that can be sent to users.


### Simple Flagging

Moderation adds a simple 'Report this Post' / 'Report this Comment' option to each post and comment.


It even keeps a handy record of warnings made against each user. Moderation is the perfect tool for growing a safe, successful, Multisite with user generated content.

### To install:

1.  Download the plugin file
2.  Unzip the file into a folder on your hard drive
3.  Upload **/moderation/** folder to **/wp-content/plugins/** folder on your site
4.  Visit **Network Admin -> Plugins** and **Network Activate** it there.

_Note: If you have an older version of the plugin installed in /mu-plugins/ please delete it._ After you Network Activate the plugin, you will see a menu appear on the **Network Admin Dashboard** with the Moderation tools! _*The Moderation Menu will also be visible in the Main Site Admin Menu as well_  To configure the settings for the Moderation Tools you will need to visit the **Network Admin>Settings** Menu   The items you input in the "Report reasons" meta boxes must be one per line.  The plugin will use these items in a dropdown to present the reporter with options of why they are reporting the post/comment/blog.

**It's VERY important to note: The moderation tools will ONLY be available on the subsites, NOT on the main site** 


changelog.txt
Plugin Name: Moderation
Author: S H Mohanjith (Incsub), Andrew Billits (Incsub), Mariusz Misiek (Incsub)

Change Log:
----------------------------------------------------------------------
----------------------------------------------------------------------

1.0.9 - 2014-09-04
----------------------------------------------------------------------
- Fix: Settings not saved with preg_split()

1.0.8.8 - 29/05/2017
----------------------------------------------------------------------
- Fixed: Updated with Dev dashboard library

1.0.8.7 - 28/05/2017
----------------------------------------------------------------------
- Fixed: PHP Fatal error with Moderation plugin (split function)
- Fixed: Removed deprecated function get_currentuserinfo()

1.0.8.6 - 22/09/2014 
----------------------------------------------------------------------
- Fixed: Confusing message about Moderation compatibility

1.0.8.5 - 01/20/2014 
----------------------------------------------------------------------
- Fixed: Cannot load moderation-post-archive.

1.0.8.4 - 12/16/2013 
----------------------------------------------------------------------
- WordPress 3.8 compatibility

1.0.8.3 - 08/21/2013 
----------------------------------------------------------------------
- Fix: archive post report query

1.0.8.2 - 05/06/2013 
----------------------------------------------------------------------
- Security improvements
- Fixed: "GROUP BY" sql error

1.0.8.1 - 05/19/2012 
----------------------------------------------------------------------
- Fixed: Reported posters get kicked to unaccessible page

1.0.8 - 04/04/2011
----------------------------------------------------------------------
- Fixed: Reported posters get kicked to unaccessible page

1.0.7 - 03/01/2011
----------------------------------------------------------------------
- Allow placement of moderation links from within a template
- More improvements

1.0.6 - 08/09/2011
----------------------------------------------------------------------
- Fixed: Call to undefined function awurlencode

1.0.5 - 08/04/2011
----------------------------------------------------------------------
- Switch to is_admin() from WP_ADMIN
- Localization improvements
- Hide moderation warning if not applicable

1.0.4 - 04/28/2011
----------------------------------------------------------------------
- Fixed: "Report This Comment" link showing in the back-end
- Use BLOG_ID_CURRENT_SITE instead of '1'

1.0.3 - 03/09/2011
----------------------------------------------------------------------
- Fixed: Moderation Plugin spam

1.0.2 - 01/10/2011
----------------------------------------------------------------------
- 3.1+ compatibility update

1.0.1 - 06/24/2009
----------------------------------------------------------------------
- Don't show "report post" link on search pages

1.0.1 - 06/24/2009
----------------------------------------------------------------------
- Don't show "report post" link on search pages

1.0.0 - 06/18/2009
----------------------------------------------------------------------
- Initial Release.


warning.php
<?php
/*
Plugin Name: Moderation
Plugin URI: 
Description:
Author: Andrew Billits (Incsub)
Version: 1.0.1
Author URI: http://incsub.com
*/

/* 
Copyright 2007-2009 Incsub (http://incsub.com)

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (Version 2 - GPLv2) as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

require_once('admin.php');
$title = __('Warnings');
$parent_file = 'warnings.php';
require_once('admin-header.php');

moderation_warnings();

include('admin-footer.php');
?>

moderation-admin.php
<?php
/*
Plugin Name: Moderation
Plugin URI: 
Description:
Author: Andrew Billits (Incsub)
Version: 1.0.1
Author URI: http://incsub.com
*/

/* 
Copyright 2007-2009 Incsub (http://incsub.com)

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License (Version 2 - GPLv2) as published by
the Free Software Foundation.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
*/

require_once('admin.php');
$post_blog_comment_count = $wpdb->get_var("SELECT COUNT(*) FROM " . $wpdb->base_prefix . "moderation_reports WHERE report_status = 'new'");
$total_count = $post_blog_comment_count;
if ($total_count > 0){
	$title = __('Moderation') . ' ' . $total_count;
} else {
	$title = __('Moderation');
}
$title = __('Moderation');
$parent_file = 'moderation-admin.php';
require_once('admin-header.php');

moderation_overview();

include('admin-footer.php');
?>